File: week01/day07/backend/schemas/schedule.py
from pydantic import BaseModel, Field


class ScheduleCreate(BaseModel):
    worker_id: int = Field(..., ge=1)
    patient_id: int = Field(..., ge=1)
    start_time: str  # ISO 8601
    end_time: str    # ISO 8601


class ScheduleUpdate(BaseModel):
    worker_id: int = Field(..., ge=1)
    patient_id: int = Field(..., ge=1)
    start_time: str
    end_time: str


class ScheduleOut(BaseModel):
    id: int
    worker_id: int
    patient_id: int
    start_time: str
    end_time: str
    status: str
    worker_name: str | None = None
    patient_name: str | None = None

    model_config = {"from_attributes": True}


class ScheduleLogOut(BaseModel):
    id: int
    schedule_id: int
    action: str
    operator_id: int
    original_worker_id: int | None = None
    new_worker_id: int | None = None
    remark: str | None = None
    created_at: str

    model_config = {"from_attributes": True}

File: week01/day07/frontend/src/types/schedule.ts
export interface Schedule {
  id: number;
  worker_id: number;
  patient_id: number;
  start_time: string;
  end_time: string;
  status: 'assigned' | 'in_progress' | 'completed' | 'cancelled';
  worker_name?: string;
  patient_name?: string;
}

export interface ScheduleLog {
  id: number;
  schedule_id: number;
  action: string;
  operator_id: number;
  original_worker_id?: number;
  new_worker_id?: number;
  remark?: string;
  created_at: string;
}

export interface ScheduleSlot {
  hour: number;
  schedule_id: number | null;
  patient_id?: number | null;
  patient_name?: string | null;
  worker_id?: number | null;
  worker_name?: string | null;
  status: string | null;
}

export interface ScheduleRow {
  worker_id?: number;
  worker_name?: string;
  patient_id?: number;
  patient_name?: string;
  slots: ScheduleSlot[];
}

export interface ScheduleViewData {
  view: 'worker' | 'patient';
  date: string;
  rows: ScheduleRow[];
}

export interface ScheduleFormData {
  worker_id: number;
  patient_id: number;
  start_time: string;
  end_time: string;
}

export interface MyScheduleItem {
  id: number;
  worker_id: number;
  patient_id: number;
  patient_name: string | null;
  start_time: string;
  end_time: string;
  status: string;
}
